src/fields/accessPromise.ts

```diff
@@ -60,7 +60,7 @@ const accessPromise = async ({
     relationshipPopulations.push(relationshipPopulationPromise({
       data,
       field,
-      depth,
+      depth: field.maxDepth || depth,
```

Comment:
If `field.maxDepth` is `0`, this will return falsey, when it should be respected. Also, I think that this should be handled _within_ the `relationshipPopulationPromise` code to be more consistent with each file's purpose.